client/client-multi/client-objective-c/Kaa/ConnectivityChecker.m

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "ConnectivityChecker.h"
+#import <SystemConfiguration/SystemConfiguration.h>
+
+@implementation ConnectivityChecker
+
+- (BOOL)isConnected {
+    BOOL returnValue = NO;
+    
+#ifdef TARGET_OS_MAC
+    
+    struct sockaddr zeroAddress;
+    bzero(&zeroAddress, sizeof(zeroAddress));
+    zeroAddress.sa_len = sizeof(zeroAddress);
+    zeroAddress.sa_family = AF_INET;
+    
+    SCNetworkReachabilityRef reachabilityRef = SCNetworkReachabilityCreateWithAddress(NULL, (const struct sockaddr*)&zeroAddress);
+    
+    
+#elif TARGET_OS_IPHONE
+    
+    struct sockaddr_in address;
+    size_t address_len = sizeof(address);
+    memset(&address, 0, address_len);
+    address.sin_len = address_len;
+    address.sin_family = AF_INET;
+    
+    SCNetworkReachabilityRef reachabilityRef = SCNetworkReachabilityCreateWithAddress(NULL, (const struct sockaddr*)&address);
+    
+#endif
+    
+    if (reachabilityRef != NULL) {
+        SCNetworkReachabilityFlags flags = 0;
+        
+        if(SCNetworkReachabilityGetFlags(reachabilityRef, &flags)) {
+            BOOL isReachable = ((flags & kSCNetworkFlagsReachable) != 0);
+            BOOL connectionRequired = ((flags & kSCNetworkFlagsConnectionRequired) != 0);
+            returnValue = (isReachable && !connectionRequired) ? YES : NO;
```

Comment:
Can we use shorter form? Something like
returnValue = isReachable && !connectionRequired;